lib/mps_statespace.h

```diff
@@ -144,6 +155,37 @@ class MPSStateSpace {
     }
   }
 
+  // Testing only. Convert the MPS to a wavefunction under "normal" ordering.
+  // Requires: wf be allocated beforehand with bd * 2 ^ nq -1 memory.
+  static void ToWaveFunction(MPS& state, fp_type* wf) {
+    const auto bd = state.bond_dim();
+    const auto nq = state.num_qubits();
+    fp_type* raw_state = state.get();
+
+    ConstMatrixMap accum = ConstMatrixMap((Complex*)(raw_state), 2, bd);
+    ConstMatrixMap next_block = ConstMatrixMap(nullptr, 0, 0);
+    MatrixMap result2 = MatrixMap(nullptr, 0, 0);
+    auto offset = 0;
+    auto result2_size = 2;
+
+    for (unsigned i = 1; i < nq - 1; i++) {
+      offset = GetBlockOffset(state, i);
+      // use of new does not trigger any expensive operations.
+      new (&next_block)
```

Comment:
I'm not familiar with this usage of `new` - is it available in C++11? (This is required for use with pybind)

I assume this is using `next_block` to store the `ConstMatrixMap` constructed here. Does this behave similarly to something like this?
```
ConstMatrixMap* next_block = new ConstMatrixMap(nullptr, 0, 0);
*next_block = ConstMatrixMap(...);
```
(apologies for errors, I've been away from C++ for a while)

Reply:
Thematically it does do something like the snippet you've shown. See the "Changing Mapped Array" section: https://eigen.tuxfamily.org/dox/group__TutorialMapClass.html . It is available in C++ 11.

Reply:
TIL "placement new" is a thing. Thanks!